guides/apm-release-process.md

```diff
@@ -0,0 +1,164 @@
+    details.
+    * execute `docker-compose up -d` to set up the docker components for testing. these are
+    primarily database endpoints.
+    * make changes and commit them.
+    * N.B. mocha and eslint are globally installed in the development environment. they are not installed by executing
+    `npm install` in the `appoptics-apm-node` root directory. if your setup varies take this into account.
+    * if your editor does integrate eslint then execute `npm run eslint`. regarding eslint - it's preferred to require
+    semi-colons but the code has substantial groups
+    * run tests `npm test`. It is often useful to run a subset of tests using the globally installed
+    mocha, e.g.,
+      - `mocha test/*.test.js` to run the basic test suite
+      - `mocha test/probes/*.test.js` to test the probes
+      - `mocha test/probes/fs.test.js` to test only the `fs` instrumentation.
+      - `test.sh` runs groups of tests and outputs a useful summary. `npm test` executes `test.sh`.
+    * install changes in local todo application using one of the two following methods and test:
+      - `npm install https://api.github.com/repos/appoptics/appoptics-apm-node/tarball/test-10.1.0-rc` where `test-10.1.0-rc`
+      is the branch to install.
+      - create a .tgz file using `npm pack` and install that via `npm install ../path/to/appoptics-apm-x.y.z.tgz`
+    * verify that the test not only apparently works but also that it creates valid traces on the collector.`env.sh`
+    provides options to test against production or staging.
+    * if the changes warrant it run more rigorous tests:
+        - the test matrix, `appoptics/node-apm-matrix`
+        - the benchmark, `appoptics/apm-benchmark-node`
+    * create PR and get it reviewed (see exceptions in REMARKS below). the previous tests can run
+    while the PR is being reviewed.
+    * when the PR has been approved take one of two steps
+      1) merge branch into master. consider squashing commits as appropriate.
+      2) perform an interactive rebase on the development machine (github doesn't implement this),
+      to a new branch, typically named x.y.z (the to-be-released version number without the 'v'
+      prefix). The reason the branch isn't named vx.y.z is to avoid conflict with the tag that
+      will be used on the release. an interactive rebase is like a selective squash where it's
+      possible to reword and combine commits, but not force all commits into one.
+          - push the new branch x.y.z via `git push -u origin x.y.z` and create a PR for that branch.
+          - this new PR will make the previous PR obsolete but verify that the two compare without differences.
+          the new PR should be exactly the same as the previous PR if the interactive rebase was done correctly;
+          only the commits (number and messages) will be changed.
+
+2. SMOKE TEST package
+
+installing a `tgz` file is about as close to doing `npm install appoptics-apm` (from the npm repository)
```

Comment:
it might be; when originally working with packagecloud i ran into problems. some were related to `appoptics-apm` and `appoptics-bindings` being private repositories to begin with. but though i don't recall the details, there was a problem i ran into that caused me to go in another direction. i don't see benefits to this approach because it's possible to create the `tgz` file or to get a github url to supply a `tgz` file, neither of which require editing `.npmrc` files.

Reply:
good to know